lib/src/hot_recharge.dart

```diff
@@ -161,4 +161,14 @@ class HotRecharge {
 
     return queryZesa;
   }
+
+  /// Query airtime balance for a specific user using their mobile number
+  Future<ApiResponse> endUserBalance (String mobileNumber) async{
+    /// If mobile number is not provided throw exception
+    if (mobileNumber.isEmpty) {
+      throw new Exception("Mobile Number cannot be empty");
```

Comment:
lets try to keep the error checking and catching in the `Api` class so that we maintain the response model of `ApiResponse`.